src/providers/mistral.ts

```diff
@@ -0,0 +1,194 @@
+        output: 0.00806 / 1000,
+      },
+    }))
+  ];
+
+  static MISTRAL_CHAT_MODELS_NAMES = MistralChatCompletionProvider.MISTRAL_CHAT_MODELS.map(
+    (model) => model.id,
+  );
+
+  constructor(
+    modelName: string,
+    options: { id?: string; config?:  MistralChatCompletionOptions; env?: EnvOverrides } = {},
+  ) {
+    if (!MistralChatCompletionProvider.MISTRAL_CHAT_MODELS_NAMES.includes(modelName)) {
+      logger.warn(`Using unknown Mistral chat model: ${modelName}`);
+    }
+    const { id, config, env } = options;
+    this.env = env;
+    this.modelName = modelName;
+    this.id = id ? () => id : this.id;
+    this.config = config || {};
+  }
+
+  id(): string {
+    return `mistral:${this.modelName}`;
+  }
+
+  toString(): string {
+    return `[Mistral Provider ${this.modelName}]`;
+  }
+
+  getApiKey(): string | undefined {
+    return this.config.apiKey || this.env?.MISTRAL_API_KEY || process.env.MISTRAL_API_KEY;
+  }
+
+  async callApi(prompt: string): Promise<ProviderResponse> {
+    if (!this.getApiKey()) {
+      throw new Error(
+        'Mistra API key is not set. Set the MISTRAL_API_KEY environment variable or add `apiKey` to the provider config.',
+      );
+    }
+
+    const messages = parseChatPrompt(prompt, [{ role: 'user', content: prompt }]);
+    
+    const body = {
+      model: this.modelName,
+      messages: messages,
+      temperature: this.config?.temperature,
+      top_p: this.config?.top_p || 1,
+      max_tokens: this.config?.max_tokens || 1024,
+      safe_prompt: this.config?.safe_prompt || false,
+      random_seed: this.config?.random_seed || null,
+    };
+
+    const url = 'https://api.mistral.ai/v1/chat/completions';
+    logger.debug(`Mistral API request: ${url} ${JSON.stringify(body)}`);
+
+    let data, 
+      cached = false;
+
+    try {
+      ({ data, cached } = (await fetchWithCache(
+        url,
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'Authorization': `Bearer ${this.getApiKey()}`,
+          },
+          body: JSON.stringify(body),
+        },
+        REQUEST_TIMEOUT_MS,
+        )) as unknown as { data: any; cached: boolean });
+    } catch (err) {
+      return {
+        error: `API call error: ${String(err)}`,
+      };
+    }
+
+    logger.debug(`Mistral API response: ${JSON.stringify(data)}`);
+
+    if (data.error) {
+      return {
+        error: `API call error: ${data.error}`,
+      };
+    }
+    if (!data.choices[0] && !data.choices[0].message.content) {
+      return {
+        error: `Malformed response data: ${JSON.stringify(data)}`,
+      };
+    }
+    let content;
+    if (this.config?.fix_json){
+      // Mistral is often messing up JSON responses. This fixes the output to allow for the tests to pass.
+      content = data.choices[0].message.content.replace(/\\_/g, "_").replace(/\\\*/g, "*")
+    } else {
+      content = data.choices[0].message.content
```

Comment:
@typpo I'm not sure if you want this kind of logic in here.
I only added it because it often causes issues with JSON responses.

See this [Reddit](https://www.reddit.com/r/LocalLLaMA/comments/1agrddy/has_anyone_encountered_mistrals_tendency_to_use/).

Reply:
I think this is ok for now if it's a common problem that everyone is working around.  If it's more niche, then it would be better handled using the [`transform` testcase option](https://promptfoo.dev/docs/configuration/guide/#transforming-outputs)